### ISSUE TITLE
rusk: `RuskState` drops `unstaged` commits too aggressively

### ISSUE THREAD

**Issue by ZER0** [dusk-network/rusk#713]
**Describe the bug**
Currently, as soon as the instance of a `RuskState` go out of scope, all the changes that are not accepted or finalized are automatically discarded.

This is intended: however, since the `network` wrapped by the `RuskState` is behind an arc mutex, this impact *all* the consumers.
To be clear: let's say we have two async call, EST and `get_anchor`.  Both requests the `RuskState`, and since they can be executed in parallel, and `get_anchor` is faster, once `get_anchor` is over it will drops all the unstaged changes made by `EST` even if `EST` is in the middle of execution.

**To Reproduce**
See: https://github.com/dusk-network/dusk-blockchain/issues/1350#issuecomment-1104818539

**Expected behaviour**
`RuskState` should lock the mutex for all the duration of its lifetime, in that way we're ensuring consistency of the data. Since, even without the dropping, if a request for the state root (read-only) is invoke during a `EST`, the `state_root` returned would be wrong.

**Logs/Screenshot**
N/A

**Platform**
N/A

**Additional context**
See: https://github.com/dusk-network/dusk-blockchain/issues/1350
